bioio/writers/ome_zarr_writer_2.py

```diff
@@ -0,0 +1,688 @@
+import logging
+from dataclasses import asdict, dataclass
+from math import prod
+from typing import Any, List, Tuple, Union
+
+import dask.array as da
+import numcodecs
+import numpy as np
+import skimage.transform
+import zarr
+from ngff_zarr.zarr_metadata import Axis, Dataset, Metadata, Scale, Translation
+from zarr.storage import DirectoryStore, FSStore, default_compressor
+
+from bioio import BioImage
+
+log = logging.getLogger(__name__)
+
+DimTuple = Tuple[int, int, int, int, int]
+
+
+def chunk_size_from_memory_target(
+    shape: DimTuple, dtype: str, memory_target: int
+) -> DimTuple:
+    """
+    Calculate chunk size from memory target in bytes.  The chunk size will be
+    determined by considering a single T and C, and subdividing the remaining
+    dims by 2 until the chunk fits within the size target.
+
+    Parameters
+    ----------
+    shape:
+        Shape of the array. Assumes a 5d TCZYX array.
+    dtype:
+        Data type of the array.
+    memory_target:
+        Memory target in bytes.
+
+    Returns
+    -------
+    Chunk size tuple.
+    """
+    if len(shape) != 5:
+        raise ValueError("shape must be a 5-tuple in TCZYX order")
+
+    itemsize = np.dtype(dtype).itemsize
+    # let's start by just mandating that chunks have to be no more than
+    # 1 T and 1 C
+    chunk_size = (1, 1, shape[2], shape[3], shape[4])
+    while prod(chunk_size) * itemsize > memory_target:
```

Comment:
I think this could be simplified to 
```   
while prod(chunk_size) * itemsize > memory_target:
    chunk_size = tuple(max(cs // 2, 1) for cs in chunk_size)
```

Reply:
nope, mypy hates that due to the enforcement of 5D tuples here.